swift/SampleApp/SampleApp/Mobility/Data/ITSMobilityService.swift

```diff
@@ -0,0 +1,89 @@
+/*
+ * Software Name : ITSClient
+ * SPDX-FileCopyrightText: Copyright (c) Orange SA
+ * SPDX-License-Identifier: MIT
+ *
+ * This software is distributed under the MIT license,
+ * see the "LICENSE.txt" file for more details or https://opensource.org/license/MIT/
+ *
+ * Software description: Swift ITS client.
+ */
+
+import Foundation
+import ITSCore
+import ITSMobility
+
+actor ITSMobilityService: MobilityService {
+    private let mobility: Mobility
+    private var roadAlarmChangeObserver: RoadAlarmChangeObserver?
+    private var sendPositionTask: Task<Void, any Error>?
+
+    init() {
+        mobility = Mobility()
+    }
+
+    func start() async throws(MobilityError) {
+        // Update this to set your broker connection configuration
+        let host = ""
+        let port = 1883
+        let clientIdentifier = UUID().uuidString
+        let username: String? = nil
+        let password: String? = nil
+        let useSSL = false
+
+        let mqttClientConfiguration = MQTTClientConfiguration(host: host,
+                                                              port: port,
+                                                              clientIdentifier: clientIdentifier,
+                                                              userName: username,
+                                                              password: password,
+                                                              useSSL: useSSL)
+        let coreConfiguration = CoreConfiguration(mqttClientConfiguration: mqttClientConfiguration)
+        let mobilityConfiguration = MobilityConfiguration(coreConfiguration: coreConfiguration,
+                                                          stationID: 165_315,
+                                                          reportZoomLevel: 22)
+        do {
+            try await mobility.start(mobilityConfiguration: mobilityConfiguration)
+        } catch {
+            throw .startFailed
+        }
+
+        let observer = await MobilityRoadAlarmChangeObserver()
+        await mobility.setRoadAlarmObserver(observer)
+        roadAlarmChangeObserver = observer
+
+        startSendingPosition()
+    }
+
+    func stop() async throws(MobilityError) {
+        stopSendingPosition()
+        do {
+            try await mobility.stop()
+        } catch {
+            throw .stopFailed
+        }
+        roadAlarmChangeObserver = nil
```

Comment:
If stop failed, why not setting observer to nil still?

Reply:
For now the subscriptions are not removed when the MQTT disconnect fail, so theoretically you can receive updates later when the network comes back for example. This behavior will be fixed in #366.  But we should ignore this error when a user wants to stop (and destroy the MQTT client even if an error occurs). With this update the stop will not throw an error anymore. If you agree with this behavior I will create a PR to fix it and we can set the observer to nil before calling stop.